### PR TITLE
feat: model primaryIdentifier in database

### PR DESCRIPTION
**Pull request by kaizencc**
Adds the `primaryIdentifier` property from the resource spec to the database. I can see the ensuing result in `db.json` after building.

It is typed as `string[]` because there are roughly 140 resources with > 1 `primaryIdentifiers`. There are also 7 resources with no `primaryIdentifiers`.

### CODE REVIEW

**Comment by kaizencc** [comment]
> Don't forget to add it to the differ :)

Thats no fair! Now i have to deal with the merge conflicts

File: packages/@aws-cdk/service-spec-build/src/resource-builder.ts [hunk 1]
```diff
@@ -11,27 +11,41 @@ import {
...
     const existing = this.db.lookup('resource', 'cloudFormationType', 'equals', typeName);
 
     if (existing.length > 0) {
       const resource = existing.only();
       if (!resource.documentation && options.description) {
         resource.documentation = options.description;
       }
+      if (!resource.primaryIdentifier) {
+        resource.primaryIdentifier = options.primaryIdentifier;
+      }
```
Comment (kaizencc) [inline packages/@aws-cdk/service-spec-build/src/resource-builder.ts:39]: I think this means that we are importing the old spec first and then the new spec, so most resources already exist in the database. Can someone confirm whether that assumption is right? Is there an equivalent property that I should be adding to the `primaryIdentifier` field from the old spec?

Comment (rix0rrr) [reply packages/@aws-cdk/service-spec-build/src/resource-builder.ts:39]: You are correct, and the old spec doesn't have this information.

All good!

File: packages/@aws-cdk/service-spec-build/src/import-cloudformation-registry.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export function importCloudFormationRegistryResource(options: LoadCloudFormation
   const specBuilder = new SpecBuilder(db);
   const resourceBuilder = specBuilder.resourceBuilder(resource.typeName, {
     description: resource.description,
+    primaryIdentifier: resource.primaryIdentifier,
```
Comment (rix0rrr) [inline packages/@aws-cdk/service-spec-build/src/import-cloudformation-registry.ts:29]: Please translate the primaryIdentifier from `/properties/WorkspaceId` to `WorkspaceId`.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Translate property names, add a test, and we're ready!

**Comment by rix0rrr** [comment]
#623 might introduce some merge conflicts, but it will make it impossible to forget to update bits

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**